expression/builtin_control.go

```diff
@@ -327,6 +330,7 @@ func (c *ifFunctionClass) getFunction(args []Expression, ctx context.Context) (s
 		return nil, errors.Trace(err)
 	}
 	retTp := c.inferType(args[1].GetType(), args[2].GetType())
+
 	evalTps := fieldTp2EvalTp(retTp)
```

Comment:
`fieldTp2EvalTp` is not tacking `tpJSON` into consideration, will we do it in the future ?

Reply:
sorry for late replay
you can add JSON in fieldTp2EvalTp as the following code:
``` sql
	case types.ClassString:
		switch tp.Tp {
		case mysql.TypeDate, mysql.TypeDatetime:
			return tpDatetime
		case mysql.TypeTimestamp:
			return tpTimestamp
		case mysql.TypeDuration:
			return tpDuration
		}
                case mysql.TypeJSON:
                       return tpJSON
	}
```